features/spicy-packet-analyzer/scripts/main.zeek

```diff
@@ -7,23 +7,23 @@ event zeek_init() &priority=20
 	# type 0x88b5. Adapt as suitable, some suggestions in comments.
 
 	# Activate our analyzer on top of Ethernet.
-	PacketAnalyzer::register_packet_analyzer(PacketAnalyzer::ANALYZER_ETHERNET, 0x88b5, PacketAnalyzer::ANALYZER_SPICY__@ANALYZER_UPPER@);
+	PacketAnalyzer::register_packet_analyzer(PacketAnalyzer::ANALYZER_ETHERNET, 0x88b5, PacketAnalyzer::ANALYZER_SPICY_@ANALYZER_UPPER@);
```

Comment:
This will break the feature for older Zeeks. Not sure what the solution is, but for now I'd prefer to keep them working with our Zeek 5.x  releases.

Reply:
The template didn't seem to work for versions before zeek-5.2.0 so I masked this here. Could you look into that?

If I run this test suite with above change reverted (also requires adjusting/removing the `@TEST-REQUIRES` clause in `tests/features/spicy-packet-analyzer`) in e.g., `zeekurity/zeek:5.1.0-rc1` I get

```console
$ btest -dv features/spicy-packet-analyzer
features.spicy-packet-analyzer ...
  > ${SCRIPTS}/zeek-version 50100
  > ${SCRIPTS}/zkg -vvvvv create --packagedir test --features=spicy-packet-analyzer --user-var name=MyPackage --user-var analyzer=MyAnalyzer --user-var unit=MyPacket
  > mkdir test/build && cd test/build && cmake .. && make -j$(nproc)
  > btest -djc test/testing/btest.cfg
... features.spicy-packet-analyzer failed
  % 'btest -djc test/testing/btest.cfg' failed unexpectedly (exit code 1)
  % cat .stderr
  2022-11-07 11:40:47 DEBUG    init Manager version 2.13.0

  ====================|  Spicy-side Zeek Installation Summary  |====================

  Found Zeek:            yes

  Version:               5.1.0-rc1 (50100)
  Prefix:                /usr/local/zeek
  Build type:            release

  ========================================================================


  ====================|  Spicy Installation Summary  |====================

  Found Spicy:           yes

  Version:               1.5.1 (10501)
  Prefix:                /usr/local/zeek
  Build type:            release
  Have toolchain:        yes
  Spicy compiler:        /usr/local/zeek/bin/spicyc

  ========================================================================

  [#1] tests.availability ... failed
    % 'zeek -NN | grep -qi 'ANALYZER_SPICY_\?MYANALYZER'' failed unexpectedly (exit code 1)
    % cat .stderr

  [#2] tests.trace ... failed
    % 'zeek -Cr ${TRACES}/raw-layer.pcap ${PACKAGE} /root/package-template/tests/.tmp/features.spicy-packet-analyzer/test/testing/.tmp/tests.trace/trace.zeek >output' failed unexpectedly (exit code 1)
    % cat .stderr
    error in /root/package-template/tests/.tmp/features.spicy-packet-analyzer/test/testing/../scripts/./main.zeek, line 14: unknown identifier PacketAnalyzer::ANALYZER_SPICY_MYANALYZER, at or near "PacketAnalyzer::ANALYZER_SPICY_MYANALYZER"

  2 of 2 tests failed

1 of 1 test failed
```

Reply:
The template in `master` is working fine for me with Zeek 5.0. 

```
# zeek -v
zeek version 5.0.2-15-debug
# ./test-spicy-features
[...]
=== TEST SUMMARY ===

--- Packet Analyzer
--- File Analyzer
--- Protocol Analyzer - UDP - one unit
--- Protocol Analyzer - UDP - two units
--- Protocol Analyzer - TCP - one unit
--- Protocol Analyzer - TCP - two units
```

Same with 5.1.

Or am I missing something?

Reply:
> The template didn't seem to work for versions before zeek-5.2.0 ...
I got confused here, before my patches it only worked for versions <zeek-5.2.0 (my patches flipped that so it now only works on >=zeek-5.2.0).

I'll try and see whether I can use some preprocessor magic to make it work against any Zeek version.